consumer-java/src/main/java/org/apache/pulsar/client/impl/weight/WeightedConsumerConfiguration.java

```diff
@@ -125,6 +124,12 @@ public int getMaxWeightAllowed() {
         return maxWeightAllowed;
     }
 
+    public String getSubscriptionName(ConsumerConfigurationData conf){
```

Comment:
this makes it seem like a config related thing, but it is rather an internal translation that the consumer should be doing, thus the logic should reside completely locally to where you are using this method.

Reply:
from a config point of view, i see 2 things wrong with this:

- gives a wrong idea about the subscription name. configured subscription name is `A/B` itself, but this method returns `A_B`
- The getters in a config class should simply be returning configured settings. This method is taking in another config object and returning a value which is not same as what's configured either in this class or in the class of the argumnet object.